src/datajudge/db_access.py

```diff
@@ -905,34 +905,28 @@ def get_column_array_agg(
 
 
 def get_ks_2sample(
-    engine: sa.engine.Engine, table1: tuple, table2: tuple
-) -> tuple[float, int, int]:
+    engine: sa.engine.Engine,
+    ref1: DataReference,
+    ref2: DataReference,
+) -> float:
     """
     Runs the query for the two-sample Kolmogorov-Smirnov test and returns the test statistic d.
     """
-
-    # make sure we have a string representation here
-    table1_selection, col1 = str(table1[0]), str(table1[1])
-    table2_selection, col2 = str(table2[0]), str(table2[1])
-
-    if is_mssql(engine):  # "tempdb.dbo".table_name -> tempdb.dbo.table_name
-        table1_selection = table1_selection.replace('"', "")
-        table2_selection = table2_selection.replace('"', "")
-
-    # for RawQueryDataSource this could be a whole subquery and will therefore need to be wrapped
-    if "SELECT" in table1_selection:
-        table1_selection = f"({table1_selection})"
-        table2_selection = f"({table2_selection})"
+    # For mssql: "tempdb.dbo".table_name -> tempdb.dbo.table_name
+    table1_str = str(ref1.data_source.get_clause(engine)).replace('"', "")
```

Comment:
Nice, didn't know you'd get the clean table name incl. parentheses `(...)` this way. Very clean!